### PR TITLE
Hc pinning from wallet

### PR DESCRIPTION
**Pull request by ThomasArts**
If external part is going to post pin then nice to know who last leader is in order to signal where the pin landed on parent chain.

This PR is supported by Æternity Foundation

### CODE REVIEW

File: apps/aecore/src/aec_pinning_agent.erl [hunk 1]
```diff
@@ -20,28 +20,46 @@
...
     {ok, BlockHash} = aec_chain_state:get_key_block_hash_at_height(First-1),
     {ok, ChainType} = aec_parent_connector:get_parent_chain_type(),
-    
-    #{epoch             => Epoch-1,
-      height            => First-1,
-      block_hash        => BlockHash,
-      parent_type       => ChainType,
-      parent_network_id => aec_parent_connector:get_network_id()}.
+    ConMod = aec_parent_connector:get_parent_conn_mod(),
+    {_,Type} = lists:split(8,atom_to_list(ConMod)), % split off "aehttpc_" from mod name to get type
```
Comment (hanssv) [inline apps/aecore/src/aec_pinning_agent.erl:32]:
```suggestion
   "aehttpc_" ++ Type = atom_to_list(ConMod),
```

Comment (klercker) [reply apps/aecore/src/aec_pinning_agent.erl:32]: or just use the the parent_connector:get_chain_type() implemented in #4430, it´s on `master`

Comment (ThomasArts) [reply apps/aecore/src/aec_pinning_agent.erl]: Removed this altogether, since now type is known by parent connector.

**Review by klercker — APPROVED** [review]

File: apps/aecore/src/aec_pinning_agent.erl [hunk 2]
```diff
@@ -20,28 +20,44 @@
...
+    end.
 
 
 -spec encode_pin_payload(#{epoch => integer(), height => integer(), block_hash => binary()}) -> binary().
 encode_pin_payload(#{epoch := Epoch, height := Height, block_hash := BlockHash}) ->
-    <<Epoch:64/integer, Height:64/integer, BlockHash/binary>>.
+    EpochHex = list_to_binary(erlang:integer_to_list(Epoch, 16)),
+    HeightHex = list_to_binary(erlang:integer_to_list(Height, 16)),
+    EncBlockHash = aeser_api_encoder:encode(key_block_hash, BlockHash),
+    <<EpochHex/binary, ":", HeightHex/binary, " ", EncBlockHash/binary>>.
```
Comment (hanssv) [inline apps/aecore/src/aec_pinning_agent.erl:53]: Shouldn't this be a proper serialization? Wasn't there a size limitation (on Bitcoin), then a colon, a space and an API-encoded thing can't be correct?

**Review by hanssv — APPROVED** [review]

File: apps/aecore/src/aec_pinning_agent.erl (hunk 2 above)
Comment (hanssv) [reply apps/aecore/src/aec_pinning_agent.erl:53]: Let's worry about that if it turns out a problem

Comment (ThomasArts) [reply apps/aecore/src/aec_pinning_agent.erl:53]: Point made: if parent chain can recognise what is posted, these transaction can be sensored. Possibly a valid point.
In this solution we made it even human readable, so certainly filterable.
Other point made: more streamline if we would serialize in aeser_api_encode for this and have a special type `hc_pin` for this. That is indeed more coherent. 

I made an issue: https://github.com/aeternity/aeternity/issues/4435

**Pull request merged**